### PR TITLE
Sort catalog filters and other enhancements

### PR DESCRIPTION
**Pull request by divyanshiGupta**
**Fixes:** https://issues.redhat.com/browse/ODC-6491

This PR adds the fix for sorting catalog filters alphabetically and 
uses useK8sWatchResource instead of k8sGet for fetching the bindable-services.

**Screenshots:**
Screenshot 2022-03-08 at 11 13 24 PM

Screenshot 2022-03-08 at 11 13 40 PM

Screenshot 2022-03-09 at 4 12 08 PM
Screenshot 2022-03-09 at 4 19 34 PM

/kind feature

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
@beaumorley can you PTAL at this PR and approve if the badge and filters looks good to you?

**Comment by christianvogt** [comment]
@beaumorley you'll see how there is a `bindable` as well as `not bindable` filter in the catalog list. This is the way the filters currently work in that all items must be covered by at least one filter. If we want to change this for boolean type filters, then we should raise a new issue with the details and not hold up this one.

**Review by jerolimov — CHANGES_REQUESTED** [review]
Get this react error and then a white page when open the OperatorBackedService catalog with this PR. Can you take a look?

```
Error
Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
Call Stack
 checkForNestedUpdates
  vendors~main-fde4aab59bedd45747de.js:465232:15
 scheduleUpdateOnFiber
  vendors~main-fde4aab59bedd45747de.js:463264:3
```

/hold

File: frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogFilters.tsx
Comment (jerolimov) [inline frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogFilters.tsx:29]: "I like types" :smirk: without defining it at the reduce or the initial parameter of the reduce its just `{}`

```suggestion
    .reduce<CatalogFilters>((acc, groupName) => {
```

File: frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogFilters.tsx
Comment (jerolimov) [inline frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogFilters.tsx:57]: Types please :)

File: frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogFilters.tsx
Comment (jerolimov) [inline frontend/packages/console-shared/src/components/catalog/catalog-view/CatalogFilters.tsx:34]: Can you add types here? :)

**Review by jerolimov — CHANGES_REQUESTED** [review]
Some more requests to add some missing types.

**Comment by serenamarie125** [comment]
Couple comments here

1. I think we need to reconsider the filter "text" here.  Let's cover it in the Friday call?  (this shouldn't hold up merging this PR though, we can have a follow PR for text as needed)
1. Shouldn't there be more spacing under the `Service Binding` title?

Otherwise looks great! FYI @slemeur

**Comment by divyanshiGupta** [comment]
> Get this react error and then a white page when open the OperatorBackedService catalog with this PR. Can you take a look?
> 
> ```
> Error
> Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
> Call Stack
>  checkForNestedUpdates
>   vendors~main-fde4aab59bedd45747de.js:465232:15
>  scheduleUpdateOnFiber
>   vendors~main-fde4aab59bedd45747de.js:463264:3
> ```
> 
> /hold

@jerolimov I couldn't reproduce this issue. Can you share the steps to reproduce this?


https://user-images.githubusercontent.com/20724543/157652299-88389415-733b-46a3-9421-b716f817e60a.mov

**Review by serenamarie125 — COMMENTED** [review]
great work!

**Comment by jerolimov** [comment]
The updated flags names fix the crash when the operator is not installed.
/unhold

**Review by jerolimov — APPROVED** [review]
Tested it locally and the code looks good to me. The broken tests are not related to this PR.

/lgtm

**Comment by invincibleJai** [comment]
@divyanshiGupta something looks off with test, PTAL

**Comment by invincibleJai** [comment]
/label docs-approved
/label px-approved

propagating above labels based on ACKs on epic

cc @rishumehra

**Comment by jerolimov** [comment]
> 2. Shouldn't there be more spacing under the `Service Binding` title?

@divyanshiGupta FYI: I created this issue and PR for the PF catalog extension.

* https://github.com/patternfly/patternfly-react/issues/7059
* https://github.com/patternfly/patternfly-react/pull/7060

**Pull request merged**
